packages/segmentedControl/components/SegmentedControlButton.tsx

```diff
@@ -36,21 +36,21 @@ export interface SegmentedControlButtonProps {
    * Content that appears in a Tooltip when a users hovers the button
    */
   tooltipContent?: React.ReactNode;
+  children?: React.ReactNode;
 }
 
-class SegmentedControlButton extends React.PureComponent<SegmentedControlButtonProps> {
-  placeholderId = nextId("segmentedControlButton-");
+const placeholderId = nextId("segmentedControlButton-");
 
-  render() {
-    const {
-      id = this.placeholderId,
-      isActive,
-      onChange,
-      name,
-      value,
-      tooltipContent,
-      children
-    } = this.props;
+const SegmentedControlButton = React.memo(
```

Comment:
How come you've added a React.memo here?

Reply:
React.memo is meant to be an alternative when refactoring a class component using [React.PureComponent](https://reactjs.org/docs/react-api.html#reactpurecomponent) to a functional component with hooks.

Reply:
I am not sure I like the idea of just trying to memoize components across the board. There is a performance hit to memoization. That being said, I can find articles [that say that you should](https://attardi.org/why-we-memo-all-the-things/), and articles [that say that you shouldn't](https://dmitripavlutin.com/use-react-memo-wisely/). In my mind, we know there is a performance hit to memoizing a component and it might not be a good idea to do it for these smaller components.

Reply:
agree that not everything should be auto-memoed. However in this case, have to maintain memoization to maintain the existing functionality